src/utils/sort.js

```diff
@@ -1,24 +1,24 @@
-export function sortDeviceInfos(deviceInfos, sortTokens) {
-  if (sortTokens[0] === "name") {
-    if (sortTokens[1] === "desc") {
-      deviceInfos.sort((a, b) => {
-        return b.name.localeCompare(a.name);
-      });
-    } else {
-      deviceInfos.sort((a, b) => {
-        return a.name.localeCompare(b.name);
-      });
-    }
-  }
-  if (sortTokens[0] === "updatedTime") {
-    if (sortTokens[1] === "desc") {
-      deviceInfos.sort((a, b) => {
-        return b.updatedTime > a.updatedTime ? 1 : -1;
-      });
-    } else {
-      deviceInfos.sort((a, b) => {
-        return a.updatedTime > b.updatedTime ? 1 : -1;
-      });
-    }
+const sortPrototypes = {
+  name: {
+    asc: (a, b) => {
+      return a.name.localeCompare(b.name);
+    },
+    desc: (a, b) => {
+      return b.name.localeCompare(a.name);
+    },
+  },
+  updatedTime: {
+    asc: (a, b) => {
+      return a.updatedTime > b.updatedTime ? 1 : -1;
+    },
+    desc: (a, b) => {
+      return b.updatedTime > a.updatedTime ? 1 : -1;
+    },
+  },
+};
+
+export function sortByElementProperty(unsortedArray, sortTokens) {
+  if (sortTokens.length === 2) {
```

Comment:
What does "2" stand for?

Reply:
it prevents error when sortTokens doesn't have anything token to select sort prototype.